src/uct/ib/base/ib_md.c

```diff
@@ -207,6 +232,43 @@ static ucs_stats_class_t uct_ib_md_stats_class = {
 };
 #endif
 
+#ifndef HAVE_TL_DC
+#  define uct_tl_dc_mlx5_init()
+#  define uct_tl_dc_mlx5_cleanup()
+#endif
+
+#ifndef HAVE_TL_RC
+#  define uct_tl_rc_verbs_init()
+#  define uct_tl_rc_verbs_cleanup()
+#endif
+
+#if !defined (HAVE_MLX5_HW) || !defined (HAVE_TL_RC)
+#  define uct_tl_rc_mlx5_init()
+#  define uct_tl_rc_mlx5_cleanup()
+#endif
+
+#ifndef HAVE_TL_UD
+#  define uct_tl_ud_verbs_init()
+#  define uct_tl_ud_verbs_cleanup()
+#endif
+
+#if !defined (HAVE_MLX5_HW_UD) || !defined (HAVE_TL_UD)
+#  define uct_tl_ud_mlx5_init()
+#  define uct_tl_ud_mlx5_cleanup()
+#endif
+
+#ifndef HAVE_MLX5_DV
+#  define uct_ib_md_ops_dv_init()
+#endif
+
+#if !defined (HAVE_MLX5_DV) || !defined (HAVE_DEVX)
+#  define uct_ib_md_ops_devx_init()
+#endif
+
+#if !defined (HAVE_MLX5_HW) || !defined (HAVE_VERBS_EXP_H)
+#  define uct_ib_md_ops_exp_init()
+#endif
```

Comment:
can it be done together with includes to have a single place of TLs existence parsing?

Reply:
> can it be done together with includes to have a single place of TLs existence parsing?

mix include and define? codestyle?

Reply:
maybe define empty ctor/dtor in C files?